llvm/lib/Transforms/Utils/CloneFunction.cpp

```diff
@@ -147,8 +147,10 @@ void llvm::CloneFunctionAttributesInto(Function *NewFunc,
 DISubprogram *llvm::CollectDebugInfoForCloning(const Function &F,
                                                CloneFunctionChangeType Changes,
                                                DebugInfoFinder &DIFinder) {
-  // CloneModule takes care of cloning debug info.
-  if (Changes == CloneFunctionChangeType::ClonedModule)
+  // CloneModule takes care of cloning debug info for ClonedModule. Cloning into
+  // DifferentModule is taken care of separately in ClonedFunctionInto as part
+  // of llvm.dbg.cu update.
+  if (Changes >= CloneFunctionChangeType::DifferentModule)
```

Comment:
FYI the check in CloneFunctionInto is:

```
  if (Changes != CloneFunctionChangeType::DifferentModule)
    return;
```

Should we make it `<` instead?

Reply:
@felipepiovezan sorry I didn't get that. Could you elaborate please?

Reply:
Given that FindDebugInfoToIdentityMap and CollectDebugInfoForCloning get removed later in the stack, I'm going to go ahead and merge this. Please, let me know if I missed something important and I'll fix it on top.